extensions/wikia/Recirculation/RecirculationHooks.class.php

```diff
@@ -61,18 +63,17 @@ public static function onOasisSkinAssetGroups( &$jsAssets ) {
 	/**
 	 * Return whether we're on one of the pages where we want to show the Recirculation widgets,
 	 * specifically File pages, Article pages, and Main pages
+	 *
 	 * @return bool
 	 */
 	static public function isCorrectPageType() {
 		$wg = F::app()->wg;
 
-		$showableNameSpaces = array_merge( $wg->ContentNamespaces, [ NS_FILE ] );
+		$showableNameSpaces = array_merge( $wg->ContentNamespaces, [ NS_FILE, NS_BLOG_ARTICLE ] );
 
-		if ( $wg->Title->exists() &&
-		     in_array( $wg->Title->getNamespace(), $showableNameSpaces ) &&
-		     $wg->request->getVal( 'action', 'view' ) === 'view' &&
-		     $wg->request->getVal( 'diff' ) === null &&
-		     !WikiaPageType::isCorporatePage()
+		if ( ( $wg->Title->exists() && in_array( $wg->Title->getNamespace(), $showableNameSpaces ) ) &&
```

Comment:
Agreed.

For the first variable, I'd go with isInShowableNamespace or something similar.
For the second variable, I'd consider naming it something like isPureViewAction so that we know we're targetting requests in viewing mode with no diff.